python/Ganga/Core/GangaRepository/GangaRepositoryXML.py

```diff
@@ -225,7 +225,7 @@ def shutdown(self):
         from Ganga.Utility.logging import getLogger
         logger = getLogger()
         logger.debug("Shutting Down GangaRepositoryLocal: %s" % self.registry.name)
-        for k in self._fully_loaded.keys():
+        for k in [_k for _k in self._fully_loaded.keys() if _k not in self.incomplete_objects]:
```

Comment:
I don't know it's a belt and braces check for now anyway, this may get relaxed.

this is just a short PR to fix the problems I'm seeing and see if they make worse ones (which given there's almost certainly a missing import/typo, these changes do)